cpp/dolfinx/common/TimeLogger.h

```diff
@@ -35,29 +33,26 @@ class TimeLogger
   ~TimeLogger() = default;
 
   /// Register timing (for later summary)
-  void register_timing(std::string task, double wall, double user,
-                       double system);
+  void register_timing(std::string task, double wall);
 
   /// Return a summary of timings and tasks in a Table
-  Table timings(std::set<TimingType> type);
+  Table timings() const;
 
   /// List a summary of timings and tasks. Reduction type is
   /// printed.
   /// @param comm MPI Communicator
-  /// @param type Set of possible timings: wall, user or system
   /// @param reduction Reduction type (min, max or average)
-  void list_timings(MPI_Comm comm, std::set<TimingType> type,
-                    Table::Reduction reduction);
+  void list_timings(MPI_Comm comm, Table::Reduction reduction) const;
 
-  /// Return timing
+  /// @brief Return timing.
   /// @param[in] task The task name to retrieve the timing for
   /// @returns Values (count, total wall time, total user time, total
   /// system time) for given task.
-  std::tuple<int, double, double, double> timing(std::string task);
+  std::pair<int, double> timing(std::string task) const;
```

Comment:
See my comment below. A pluralised version that returns the entire std::map would be useful.